aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -101,7 +100,16 @@ SparseTensor new_with_dims_and_tensor_sparse(
     const TensorOptions& options) {
   SparseTensor self = new_sparse(options);
   get_sparse_impl(self)->resize_(sparse_dim, dense_dim, size);
-  alias_into_sparse(self, indices, values);
+  // NOTE: There is no guarantee that `indices` and `values` don't contain AutogradMeta. However,
+  // we want to maintain the invariant that `indices_` and `values_` of a sparse tensor don't
+  // contain AutogradMeta, and to achieve that we shallow-copy `indices` and `values` here.
+  auto indices_shallow_copy = LongTensor(indices.unsafeGetTensorImpl()->shallow_copy_and_detach(
+    /*version_counter=*/indices.unsafeGetTensorImpl()->version_counter(),
+    /*allow_tensor_metadata_change=*/true));
+  auto values_shallow_copy = Tensor(values.unsafeGetTensorImpl()->shallow_copy_and_detach(
+    /*version_counter=*/values.unsafeGetTensorImpl()->version_counter(),
+    /*allow_tensor_metadata_change=*/true));
```

Comment:
We don't set `allow_tensor_metadata_change` to false here, because we need to be able to resize the sparse tensor after construction, e.g. in https://github.com/pytorch/pytorch/blob/master/test/test_sparse.py#L1852.